bindings/js/api.js

```diff
@@ -1,76 +1,150 @@
+          throw Error("in " + obj_name + ", argument #" + i + " is a buffer and should have a 'size' field");
+        if (arg.kind === "input" && arg.type.startsWith("buffer") && !("interface_index" in arg))
+          throw Error("in " + obj_name + ", argument #" + i + " is an input and should have a 'interface_index' field");
+        if ((arg.kind === "output" || (arg.kind === "input" && arg.interface_index !== undefined)) && arg.tests === undefined)
           throw Error("please provide a 'tests' field for argument #" + i + " of " + obj_name + " in api.json");
-        }
-        if ((arg.kind === "output" || (arg.kind === "input" && arg.interface_index !== undefined)) && !Array.isArray(arg.tests)) {
+        if ((arg.kind === "output" || (arg.kind === "input" && arg.interface_index !== undefined)) && !Array.isArray(arg.tests))
           throw Error("the 'tests' field for argument #" + i + " of " + obj_name + " should be an array");
-        }
-        if (arg.type === "int" && arg.kind === "input") {
+
+        if (arg.type === "int32" && arg.kind === "input")
           length_args_available.push(arg.name);
-        }
+        if (arg.type.startsWith("buffer") && arg.kind === "input" && typeof arg.size === "string")
+          // TODO: we should first check the field is syntactically correct
+          // before parsing it... if parsing fails, this will yield a
+          // nonsensical size and the error will be caught below.
+          length_args_available.push(parseSize(arg.size)[0]);
       });
-      func_obj.args.map(function(arg, i) {
-        if (arg.type === "buffer" && typeof arg.size === "string") {
-          if (!length_args_available.includes(arg.size) &&
-              !length_args_available.includes(arg.size.substring(0, arg.size.indexOf("+"))) &&
-              !length_args_available.includes(arg.size.substring(0, arg.size.indexOf("-")))) {
-            throw Error("incorrect 'size' field value (" + arg.size + ")for argument #" + i + " of " + obj_name + " in api.json");
-          }
-        }
+      func_obj.args.forEach(function(arg, i) {
+        if (arg.type.startsWith("buffer") && typeof arg.size === "string" && arg.kind === "output" &&
+          !length_args_available.includes(arg.size) &&
+          !length_args_available.includes(arg.size.substring(0, arg.size.indexOf("+"))) &&
+          !length_args_available.includes(arg.size.substring(0, arg.size.indexOf("-"))) &&
+          !length_args_available.includes(arg.size.substring(0, arg.size.indexOf("/"))) &&
+          !length_args_available.includes(arg.size.substring(0, arg.size.indexOf("*")))
+        )
+          throw Error("incorrect 'size' field value (" + arg.size + ")for argument #" + i + " of " + obj_name + " in api.json");
       });
       if (func_obj.return === undefined) {
         throw Error("please provide a 'return' field for " + obj_name + " in api.json");
       }
-    });
-  });
+    };
+  };
 };
-validateJSON(api_json);
```

Comment:
I suspect you moved this validation call somewhere else ?

Reply:
yes, I moved it because it's now a promise being resolved here: https://github.com/project-everest/hacl-star/pull/547/files#diff-ef3f015ccaf90df83df03c4a14fb05bde98677501f5524461f7684d415bcebb1R619

in the case of a web-based setting, you use `fetch` to retrieve the JSON off the network, so it's not immediately available